Storage/Bookmarks/Bookmarks.swift

```diff
@@ -14,6 +14,10 @@ public protocol SearchableBookmarks: class {
     func bookmarksByURL(_ url: URL) -> Deferred<Maybe<Cursor<BookmarkItem>>>
 }
 
+public protocol CountableBookmarks: class {
+    func numberOfBookmarks() -> Deferred<Maybe<Int>>
```

Comment:
> Is it counting bookmarks created on this device, or anywhere? Only bookmarks in the Mobile Bookmarks folder? Does it include folders, livemarks, separators? 

Probably the best way to put it is all of the bookmarks the user has read access to on their mobile device. For example, when we generate the highlights for Activity Stream, we use `ViewBookmarksLocalOnMirror` to give us all of the reconciled bookmarks a user has. This will exclude any bookmarks that are in the buffer but we only generate highlights on the user's view of the bookmark world. I suppose that would also include livemarks and separators as those are stored as items which is fine I think.

> Further, this will be sufficient to uniquely identify some users. What approach are you taking to anonymize these numbers?

Good question. Do we handle this on desktop Activity Stream? /cc @sarracini @ncloudioj